server/build-plugins.js

```diff
@@ -47,11 +47,21 @@ async function installRemotePackages(installPackages) {
         return str;
       });
       logger.info(`Installing plugin packages: ${pkgStr.split(' ')}`);
-      execSync(`npm install --no-save ${pkgStr} --production`, {
-        stdio: [0, 1, 2],
-        cwd: resolve(__dirname, '..')
-      });
-      logger.info('Done installing remote plugins');
+
+      try {
+        await execSync(`npm install --no-save ${pkgStr} --production`, {
+          stdio: [0, 1, 2],
+          cwd: resolve(__dirname, '..')
+        });
+        logger.info('Done installing remote plugins');
+      } catch (e) {
+        logger.error(
+          'There was an error installing plugins. Sometimes this is because of permissions errors \
+in node_modules. Try deleting the node_modules directory and running `npm install` again.'
+        );
+        logger.error(e.stack);
+        process.exit(1);
```

Comment:
@tynes I was running into an issue here that I think you might be well positioned to help with. Basically there's a weird edge case where previous plugins installed into node_modules can't be overwritten by the child_process `npm install` due to a `checkPermissions` warning. The warning above addresses the issue, however, as this is launched by another parent process, webpack continues to run its `watch` and `build` even after this child process exits. Ideally this will just fail up and stop everything so that the issue can be addressed.

`build-plugins.js` is run by a Webpack plugin that can run shell scripts [here](https://github.com/bpanel-org/bpanel/blob/development/configs/webpack.config.js#L162), and I'm hoping that this can be fixed without having to fork that project or create our own to run scripts before each build. I tried adding `|| { echo 'my_command failed' ; exit 1; }` after the shell command but sadly that didn't seem to stop the webpack process after failure. 

Unfortunately this is kind of hard to reproduce. It occurred after I ran a new build with a fresh zip of the repo, and ran everything with the old .bpanel directory, and then moved that, ran `npm install` again to build a new config dir, and ran the run script again at which point there were the permission issues. Clearing the problem directories and running again fixed things. Presumably restrictive permissions on a plugin dir in node_modules though would result in the same thing.

Reply:
Are you saying that you want webpack to exit if this script fails?

Reply:
Exactly, it should fail out of the build if it can't install the modules for whatever reason because the build will encounter problems anyway. It's not that the build process will fail and exit if the `npm install` doesn't work, but the build itself won't build correctly so we should fail out so that the problem can be fixed.

Reply:
Looking at their source code,

https://github.com/1337programming/webpack-shell-plugin/blob/master/lib/index.js#L203
https://nodejs.org/api/child_process.html#child_process_event_close

It looks like they catch the error and throw if it is non-zero

Reply:
This feels really hacky but it would be possible to say write a `pid` file to `/tmp` on bpanel startup and then read the `pid` file and pass the value to https://nodejs.org/api/process.html#process_process_kill_pid_signal

Not sure if this would work on windows though

Reply:
I'm not seeing where it's thrown? Either way, this never gets sent to up to the process that runs webpack in index.js.